dask_match/repartition.py

```diff
@@ -0,0 +1,53 @@
+import functools
+
+from dask.dataframe import methods
+
+from dask_match.expr import Expr
+
+
+class Repartition(Expr):
+    # TODO: Make this a proper abstract expression
+    pass
```

Comment:
I'm glad to see some repartitioning.  This should be, I hope, useful to test out some structures like `Partitions` and culling generally.

If you don't have them already (forgive me if I've missed them) can I ask for a few tests around this functionality?

Reply:
Yes, I agree with this. I will probably break this into a separate PR today and add test coverage.